src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5803,6 +5811,62 @@ public enum EventManifestOptions
         /// </summary>
         AllowEventSourceOverride = 0x8,
     }
+#if FEATURE_CORECLR && FEATURE_PAL   
+    internal  class EventLogger : EventListener
+    {
+        public EventLogger() {  }
+
+        private static string Serialize(ReadOnlyCollection<string> payloadname, ReadOnlyCollection<object> payload, string sep = ", ")
+        {
+
+            StringBuilder res = new StringBuilder();
+            var fielditer = payloadname.GetEnumerator();
+            var valueiter = payload.GetEnumerator();
```

Comment:
Seems like a fair amount of allocations to be doing while tracing.  Consider trimming this down, e.g. using StringBuilderCache if we expect most of the strings to fall below its threshold, using ReadOnlyCollection's indexer rather than allocating multiple enumerator objects, etc.